### PR TITLE
Feature/sortable monthly stats

### PR DESCRIPTION
**Pull request by Veith3n**

### CODE REVIEW

File: app/controllers/trips_controller.rb [hunk 1]
```diff
@@ -1,21 +1,53 @@
 class TripsController < ApplicationController
+  skip_before_action :verify_authenticity_token
+
   def new
     @trip = Trip.new
   end
 
   def create
-    @trip = Trip.create(trip_params)
+    @trip = Trip.new(trip_params)
```
Comment (ulska) [inline app/controllers/trips_controller.rb]: Are you sure you need instance variable here?

File: app/models/trip.rb [hunk 2]
```diff
@@ -5,6 +5,7 @@ class Trip < ApplicationRecord
 
   scope :weekly, -> { where(date: Date.current.beginning_of_week..Date.current.end_of_week).order(:date) }
   scope :monthly, -> { where(date: Date.current.beginning_of_month..Date.current.end_of_month).order(:date) }
+  scope :monthly_grouped_by_date, -> { monthly.group_by(&:date) }
```
Comment (ulska) [inline app/models/trip.rb]: Scopes shouldnt be responsible for grouping.

File: spec/factories/trips.rb [hunk 3]
```diff
@@ -1,7 +1,7 @@
 FactoryBot.define do
   factory :trip do
-    sequence(:start_address) { |n| "Miodowa #{n}, Warszawa" }
-    sequence(:destination_address) { |n| "Nowogordzka #{n}, Warszawa" }
+    start_address 'Miodowa 1, Warszawa'
```
Comment (ulska) [inline spec/factories/trips.rb]: :+1:

File: app/controllers/trips_controller.rb [hunk 4]
```diff
@@ -1,21 +1,53 @@
...
-    render json: TripStats.weekly_stats.to_json
+    stats = TripStats.new.weekly_stats
+    render json: TripPresenter.weekly_stats_serialized(stats).to_json
   end
 
   def monthly
-    render json: TripStats.monthly_stats.to_json
+    if order_param_correct? && order_type_correct?
+      orderParam = sort_params[:orderParam]
+      orderType = sort_params[:orderType]
```
Comment (ulska) [inline app/controllers/trips_controller.rb:26]: Is it neccessary to extract `sort_params[:orderParam]` and `sort_params[:orderType]` to separate variables?

File: spec/controllers/trips_controller_spec.rb [hunk 5]
```diff
@@ -16,4 +16,20 @@
...
     end
   end
+
+  context '#monthly' do
+    it 'returns json when provided params are incorrect' do
+      params = { orderParam: 'day', orderType: 'incorrect_param' }
+
+      get :monthly, params: params
+
+      expect(response).to have_http_status(422)
```
Comment (ulska) [inline spec/controllers/trips_controller_spec.rb:26]: You should also check the error message, not only status.

File: app/controllers/trips_controller.rb (hunk 4 above)
Comment (aarek) [reply app/controllers/trips_controller.rb:26]: Use `snake_case` instead of `camelCase` for variables names ;)

**Pull request merged**